src/nft/utils/asset.ts

```diff
@@ -44,3 +45,36 @@ export const generateTweetForPurchase = (assets: UpdatedGenieAsset[], txHashUrl:
   } with @Uniswap 🦄\n\nhttps://app.uniswap.org/#/nfts/collection/0x60bb1e2aa1c9acafb4d34f71585d7e959f387769\n${txHashUrl}`
   return `https://twitter.com/intent/tweet?text=${encodeURIComponent(tweetText)}`
 }
+
+function getMinListingPrice(listings: Listing[]): number {
+  return min(listings.map((listing) => listing.price ?? 0)) ?? 0
+}
+
+function mapAssetsToCollections(assets: WalletAsset[]): { collection: string; items: string[] }[] {
+  const collections = assets.map((asset) => asset.collection?.twitterUrl ?? asset.collection?.name ?? '')
+  const uniqueCollections = [...new Set(collections)]
+  return uniqueCollections.map((collection) => {
+    return {
+      collection,
+      items: assets
+        .filter((asset) => asset.collection?.twitterUrl === collection || asset.collection?.name === collection)
+        .map((asset) => asset.name ?? ''),
+    }
+  })
+}
+
+export const generateTweetForList = (assets: WalletAsset[]): string => {
+  const tweetText =
+    assets.length == 1
+      ? `I just listed ${assets[0].collection?.twitterUrl ? `${assets[0].collection?.twitterUrl} ` : ''}${
```

Comment:
you should check if the twitterUrl is defined or else it returns null for instance LilNouns:
<img width="320" alt="Screenshot 2023-02-04 at 2 33 20 PM" src="https://user-images.githubusercontent.com/9094792/216790860-c46a3490-3b08-4107-9717-91cca4034819.png">

Reply:
Great catch, this was actually an issue not with the tweet generator but how we parsed in the twitterName from gql. Fixed now.